### PR TITLE
Not supported repo providers 

### PR DESCRIPTION
**Pull request by kanika1894**
Since we support tfs and gitHub repo providers, this will let us improve insights on the failures due to this issue.
Though the status of pipeline will be failed (since the flow is breaking in between), the whitlelisted key will help us have better analysis.

### CODE REVIEW

File: src/configure/configure.ts [hunk 1]
```diff
@@ -405,16 +405,20 @@ class Orchestrator {
...
                 }
                 else if (remoteUrl.indexOf("gitlab.com") >= 0) {
                     repositoryProvider = "GitLab";
                 }
+                else {
+                    repositoryProvider = remoteUrl;
+                }
 
                 telemetryHelper.setTelemetry(TelemetryKeys.RepoProvider, repositoryProvider);
+                telemetryHelper.setTelemetry(TelemetryKeys.IsErrorWhitelisted, "true");
```
Comment (vineetmimrot) [inline src/configure/configure.ts:421]: You should also put a if condition for IsErrorWhitelisted on line 66 and don't set the result to failed.

Comment (kanika1894) [reply src/configure/configure.ts:421]: So basically have a variable IsErrorWhitelisted and put condition according to that?
(I though our query for results will have the condition based on the telemetry key)

Comment (vineetmimrot) [reply src/configure/configure.ts:421]: Yes have a variable and put failed/succeed condition acc to that and also set the whitelisted telemetry values. So by default we will get all the whitelisted error on succeeded and if we need to dig down further in future we can use the isWhitelisted flag in kusto.

Comment (kanika1894) [reply src/configure/configure.ts:426]: updated

File: src/configure/configure.ts [hunk 2]
```diff
@@ -64,9 +64,13 @@ export async function configurePipeline(node: AzureTreeItem) {
         }
         catch (error) {
             if (!(error instanceof UserCancelledError)) {
-                extensionVariables.outputChannel.appendLine(error.message);
-                vscode.window.showErrorMessage(error.message);
-                telemetryHelper.setResult(Result.Failed, error);
+                if (extensionVariables.isErrorWhitelisted === true) {
```
Comment (vineetmimrot) [inline src/configure/configure.ts:67]: We should show error to the user in this case also.

File: src/configure/configure.ts [hunk 3]
```diff
@@ -64,9 +64,13 @@ export async function configurePipeline(node: AzureTreeItem) {
         }
         catch (error) {
             if (!(error instanceof UserCancelledError)) {
-                extensionVariables.outputChannel.appendLine(error.message);
                 vscode.window.showErrorMessage(error.message);
-                telemetryHelper.setResult(Result.Failed, error);
+                if (extensionVariables.isErrorWhitelisted === true) {
+                    telemetryHelper.setResult(Result.Succeeded);
+                } else {
+                    extensionVariables.outputChannel.appendLine(error.message);
```
Comment (vineetmimrot) [inline src/configure/configure.ts:71]: what is this for? any idea?

Comment (kanika1894) [reply src/configure/configure.ts]: Not sure about the role, but since found at every place where result is being set, have made it common here as well.

**Pull request merged**